http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -153,23 +153,27 @@ void close() {
    * @param errorHandler Invoked if there is a failure to connect.
    * @throws IllegalStateException Thrown if connection is already open (eg: connect called twice).
    * @throws IllegalStateException Thrown if connection has been closed (ie: 'close()' was called)
-   * @return A {@link CompletableFuture} behaving like the {@link CompletableFuture} returned by
-   *     {@link WebSocket.Builder#buildAsync(URI, Listener)} in case of an error.
    */
-  CompletableFuture<WebSocket> connect(
+  @SuppressWarnings("FutureReturnValueIgnored")
+  void connect(final WebSocketConnectionListener listener, final Consumer<String> errorHandler) {
+    connectInternal(listener, errorHandler);
+  }
+
+  @VisibleForTesting
+  CompletableFuture<Void> connectInternal(
```

Comment:
Why are we returning a value here, just for testing maybe?

Reply:
Exactly, I wasn't sure how I could wait for the future to complete otherwise